core/Updates/3.5.1-b1.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Updates;
+
+use Piwik\Filesystem;
+use Piwik\Notification;
+use Piwik\Plugin\Manager;
+use Piwik\SettingsServer;
+use Piwik\Updater;
+use Piwik\Updates as PiwikUpdates;
+
+class Updates_3_5_1_b1 extends PiwikUpdates
+{
+    public function doUpdate(Updater $updater)
+    {
+        // try to deactivate and remove the GeoIP2 plugin from marketplace,
+        // which causes problems with GeoIp2 plugin included in core
+        // Skip for Windows as file system is case insensitive and names are handled equal
+        try {
+            if (!SettingsServer::isWindows() && is_dir(PIWIK_INCLUDE_PATH . '/plugins/GeoIP2')) {
```

Comment:
Think this also needs to check for OS X too, the update deleted the GeoIp2 plugin when I ran it locally:

![image](https://user-images.githubusercontent.com/125140/40209215-dc341968-59f2-11e8-9841-0ada3a98b5de.png)

Reply:
otherwise looks good

Reply:
Wondering how to detect if it's OS X. Maybe it would be better to find a solution to check if the file system is case insensitive. Maybe I could create a temp file with uppercase name and check if it exists with lower case name and run the update script only if that doesn't work?

Reply:
That could work. If readdir maintains case in the results, could check the file name in php. Can’t think of anything else...

Reply:
OSX can have case or case insensitive file systems AFAIK and just because it is windows doesn't mean it has some specific case. You may rather want to detect which plugin it is by checking eg for the existence of a specific file of the other plugin and this plugin etc?

Reply:
Just checking for existency won't work I think. The update will download the new version and overwrite all files with the new one. For a case insensitive filesystem that means the contents of the old GeoIP2 plugin will be overwritten with our GeoIp2 plugin. So checking for existency of any file of the old plugin can just be used to check if it was installed at all. Guess we need to check the content of specific files to be sure if it was overwritten (case insensitive) or exists in a separate directory (case sensitive)

Reply:
I've know updated the check. Checking if a file of our new `GeoIp2` plugins is located in `GeoIP2` folder should hopefully be enough to detect case insensitive file systems properly. I'm not able to test that right now, as I don't have a windows or mac setup with php. @Findus23 @tsteur @diosmosis anyone of you able to prove it doesn't remove anything on a case insensitive file system?